src/sidebar/components/tag-editor.js

```diff
@@ -1,35 +1,197 @@
     }
   };
+
+  const handleOnInput = e => {
+    if (e.inputType === 'insertText') {
+      // Show the suggestions if the user types something into the field
+      setShowSuggestions(true);
+    } else if (
+      e.inputType === undefined ||
+      e.inputType === 'insertReplacementText'
+    ) {
+      // nb. Chrome / Safari reports undefined and Firefox reports 'insertReplacementText'
+      // for the inputTyp value when clicking on an element in the datalist.
+      //
+      // There are two ways to arrive here, either click an item with the mouse
+      // or use keyboard navigation and press 'Enter'.
+      //
+      // If the input value typed already exactly matches the option selected
+      // then this Event won't fire and a user would have to press 'Enter' a second
+      // time to trigger the handleKeyPress callback above to add the tag.
+      addTag();
+    } else if (inputEl.current.value.length === 0) {
+      // If the user deleted input, hide suggestions. This has
+      // no effect in Safari and the list will stay open.
+      setShowSuggestions(false);
+    }
+  };
+
+  const suggestionsList = () => {
+    return (
+      <datalist
+        id={`tag-editor-datalist-${annotation.id}`}
+        className="tag-editor__suggestions"
+        aria-label="Annotation suggestions"
+      >
+        {showSuggestions &&
+          suggestions.map((suggestion, index) => (
+            <option key={index} value={suggestion} />
+          ))}
+      </datalist>
+    );
+  };
+
+  return (
+    <section className="tag-editor">
+      <ul
+        className="tag-editor__tag-list"
+        aria-label="Suggested tags for annotation"
+      >
+        {tagList.map(tag => {
+          return (
+            <li
+              key={`${tag}`}
+              className="tag-editor__tag-item"
+              aria-label={`Tag: ${tag}`}
+            >
+              <span className="tag-editor__edit">{tag}</span>
+              <button
+                onClick={() => {
+                  removeTag(tag);
+                }}
+                title={`Remove Tag: ${tag}`}
+                className="tag-editor__delete"
+              >
+                <SvgIcon name="cancel" />
+              </button>
+            </li>
+          );
+        })}
+      </ul>
+      <input
+        list={`tag-editor-datalist-${annotation.id}`}
+        onInput={handleOnInput}
+        onKeyPress={handleKeyPress}
+        ref={inputEl}
+        placeholder="Add tags..."
+        className="tag-editor__input"
+        type="text"
+      />
+      {suggestionsList()}
+    </section>
+  );
 }
 
-module.exports = {
-  controller: TagEditorController,
-  controllerAs: 'vm',
-  bindings: {
-    tags: '<',
-    onEditTags: '&',
-  },
-  template: require('../templates/tag-editor.html'),
+/**
+ * @typedef Tag
+ * @param tag {string} - The tag text
+ */
+
+TagEditor.propTypes = {
+  /* Annotation that owns the tags. */
+  annotation: propTypes.object.isRequired,
```

Comment:
It looks to me like the tag editor doesn't use any properties from the annotation other than its ID, so if `TagEditor` just generated its own ID it could avoid this dependency.